jenkins/scripts/cbts/coverage_utils/cbts_plugin.py

```
@@ -0,0 +1,164 @@
+    # picks it up at import time in sitecustomize.py. Order matters: env
+    # var is set BEFORE the yield, so subprocesses launched in fixture
+    # setup also see it.
+    os.environ["CBTS_TEST_ID"] = nodeid
+
+    # Sentinel is an independent "plugin fired for this nodeid" breadcrumb,
+    # used by the post-run audit to split MISSING into "plugin never ran"
+    # vs "plugin ran but data was lost". Filename = sha1(nodeid) to dodge
+    # any path-escape issues from "::" / "/" in nodeids.
+    os.makedirs(SENTINEL_DIR, exist_ok=True)
+    sentinel = os.path.join(
+        SENTINEL_DIR, hashlib.sha1(nodeid.encode()).hexdigest()
```

Ruff (0.15.15)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)